r/tests/testthat/test-data-type.R

```diff
@@ -427,3 +427,80 @@ test_that("DataType to C-interface", {
   # must clean up the pointer or we leak
   delete_arrow_schema(ptr)
 })
+
+test_that("DataType$code()", {
```

Comment:
What about a test helper here that lets you not only eliminate some copy-paste but also tests that the call that is generated is valid and does what it claims:

```r
test_code_roundtrip <- function(x) {
  expect_identical(eval(x$code()), x)
}
```

Reply:
`expect_identical()` is a bit too strong it seems but this looks fine, I'll add: 

```r
test_code_roundtrip <- function(x) {
  expect_true(eval(x$code())$Equals(x))
}
```

Reply:
FWIW that could also be written as 

```
test_code_roundtrip <- function(x) {
  expect_equal(eval(x$code()), x)
}
```

if you thought that were more readable because we use a version of expect_equal that will call that Arrow `$Equals()` method